Refactored flow for excuting in container and what is exposed in the lib

DunderRoffe opened the pull request:
This pull request refactors the flow of executing code inside of a container. The idea is that only FunctionJob and CommandJob is used for this purpose.

The second part of this pull request changes what is exposed in the library. I.e. moves header files from include to appropriate location under src.

Commit:
Refactor how SCAG create Function and Command jobs

Add two functions in SoftwareContainer: createFunctionJob and createCommandJob.

Refactor m_jobs in agent from vector to map, maps pids to CommandJobs

Refactor checkJob to getJob, returns a shared_ptr instead.

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

Commit:
Remove executeInContainer from gateway class

Remove executeInContainer(cmd) and executeInContainer(function) from the
gateway class. This change force gateway implementation to use FunctionJob
and CommandJob to execute code inside the container.

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

Commit:
Refactored what is included in the lib

Add Executable that wraps public parts of ContainerAbstractInterface
Add bindMount function to SoftwareContainer

Moved container.h, containerabstractinterface.h, gateway.h and netlink.h
from include to appropriate location under src

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

tobsan commented on libsoftwarecontainer/src/gateway/gateway.cpp:
```diff
@@ -119,6 +119,11 @@ bool Gateway::hasContainer()
 
 std::shared_ptr<ContainerAbstractInterface> Gateway::getContainer()
 {
+    if (!hasContainer()) {
```
Imho this should be an exception rather than just an error.

tobsan commented on libsoftwarecontainer/src/gateway/network/iptableentry.h:
```diff
@@ -97,6 +97,12 @@ class IPTableEntry
 
+
+    /**
+     * @brief Creates a string with infromation about the entry
```
infromation?

Commit:
Remove unused function getJob from the agent

Remove unused function getJob from the agent since it is no longer used
anywhere. It is a residue from when the dbus API supported write to stdin.

As a result of this the member m_jobs is also removed

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

Commit:
Refactor getContainer to throw exception

Refactor getContainer() in Gateway to throw an exception instead of
returning an empty shared_ptr and logging an error when called before
setContainer().

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

Commit:
Fix typo in IPTableEntry

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>
